oneandone/src/main/java/org/apache/jclouds/oneandone/rest/compute/function/ServerToNodeMetadata.java

```diff
@@ -78,26 +81,43 @@ public ServerToNodeMetadata(Function<Hdd, Volume> fnVolume,
    @Override
    public NodeMetadata apply(final Server server) {
       checkNotNull(server, "Null server");
+      String hardwareId = null;
 
       DataCenter dataCenter = api.dataCenterApi().get(server.datacenter().id());
       Location location = find(locations.get(), idEquals(dataCenter.id()));
       Hardware hardware = null;
+      Float size = null;
+      Integer minRam = 0;
+      List<Hdd> hdds = null;
+      List<Volume> volumes = Lists.newArrayList();
+      List<Processor> processors = new ArrayList<Processor>();
+
       //check if the server was built on a hardware flavour(Fixed instance)
-      if (server.hardware().fixedInstanceSizeId() != null && !"0".equals(server.hardware().fixedInstanceSizeId())) {
-         hardware = hardwareFlavors.get().get(server.hardware().fixedInstanceSizeId());
+      if (server.serverType() == Types.ServerType.baremetal) {
```

Comment:
reuse `HardwareFlavourToHardware` to get the same information, but without calling the `api` again